src/lxc/start.c

```diff
@@ -1288,28 +1286,28 @@ static int lxc_spawn(struct lxc_handler *handler)
 	}
 
 	/* Create a process in a new set of namespaces. */
-	flags = handler->clone_flags;
+	handler->on_clone_flags = handler->clone_flags;
 	if (handler->clone_flags & CLONE_NEWUSER) {
 		/* If CLONE_NEWUSER and CLONE_NEWNET was requested, we need to
 		 * clone a new user namespace first and only later unshare our
 		 * network namespace to ensure that network devices ownership is
 		 * set up correctly.
 		 */
-		flags &= ~CLONE_NEWNET;
+		handler->on_clone_flags &= ~CLONE_NEWNET;
 	}
 
-	if (fork_before_clone)
-		handler->pid = lxc_fork_attach_clone(do_start, handler, flags | CLONE_PARENT);
+	if (share_ns)
+		ret = lxc_clone(do_share_ns, handler, CLONE_VFORK | CLONE_VM | CLONE_FILES);
```

Comment:
You say "Some glibc versions used to reset the pid/tid to -1 when CLONE_VM was used without CLONE_THREAD". How old are those versions?

Reply:
Not super old. But it shouldn't matter since we don't care about getpid() in the intermediate process.

Reply:
I remember Andrei Vagin having this issue on glibc 2.24. The pid cache got removed in 2.25 I think.

Reply:
But they actually had important stuff in the child fun.